compile-release-tools

```diff
@@ -55,23 +60,27 @@ check_deps() {
   go list -u -m -json all | go-mod-outdated "${gmo_args[@]}"
 }
 
-compile_krel() {
+compile_with_flags() {
   local git_tree_state
   local pkg
+  local tool="$1"
 
-  git_tree_state=clean
+  git_tree_state=dirty
   pkg=k8s.io/release/pkg/version
 
-  if git status --porcelain --untracked-files=no; then
-    git_tree_state=dirty
+  if git_status=$(git status --porcelain --untracked=no 2>/dev/null) && [[ -z "${git_status}" ]]; then
+    git_tree_state=clean
   fi
 
-  go build -ldflags "-s -w \
+  if go build -v -ldflags "-s -w \
     -X $pkg.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ') \
     -X $pkg.gitCommit=$(git rev-parse HEAD) \
     -X $pkg.gitTreeState=$git_tree_state \
     -X $pkg.gitVersion=$(git describe --abbrev=0)" \
-    -o "$GOPATH"/bin/krel ./cmd/krel
+    -o "$RELEASE_TOOL_BIN/$tool" "./cmd/$tool"; then
+
+    echo "$tool was successfully compiled and installed to $RELEASE_TOOL_BIN/$tool"
+  fi
```

Comment:
Why do we have this else here? Isn't this hiding errors? Don't we want to bail out when any compilation fails?

Reply:
@hoegaarden -- Fixed in a follow-up. There wasn't an `else` statement, but yes, it's possible that an error could've been swallowed in that clause. Thanks!